### PR TITLE
24 full gc query functionality

### LINKED ISSUE #80: Debug transfer following query PR merge
## Description

The transfer unit test did not catch the issues present during the demo after the query PR was merged, go back and fix and add new unit tests to make sure this does not happen again.

## Dependencies

- with third parties, data etc.

## Tests

- [] What tests are needed to accomplish the success criteria

## Success Criteria

- [] Checkboxes describing every step needed to be taken for defining this ticket as Done.

### PR DESCRIPTION
**Pull request by CAGalbraith**
## Description

Expands the query function to support all combinations of parameters defined in the `GranularCertficateAction` data model, with additional unit tests and validation on the lifecycle actions that use the query as an entry point.
- Allows fractional transfers and cancellations by specifying either a bundle quantity or percentage.
- By default, queries will never return bundles marked as soft deleted. Those deleted during the bundle splitting process are marked with a `Bundle Split` status to indicate the reason for their deletion.
- Adds a `main` endpoint to change the logging level during run time for diagnosis/demoing.

Will be superceded to reflect more realistic workflows where we divide each action into a distinct data model with the minimum parameter set needed.

Closes #80 

## Has This Been Tested?

- [ ] Yes

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: gc_registry/certificate/routes.py [hunk 1]
```diff
@@ -135,6 +136,12 @@ def certificate_bundle_cancellation(
     esdb_client: EventStoreDBClient = Depends(events.get_esdb_client),
 ):
     """Cancel a fixed number of certificates matched to the given filter parameters within the specified Account."""
+
+    # If no beneficiary is specified, default to the account holder
+    if certificate_bundle_action.beneficiary is None:
```
Comment (ClemAtt) [inline gc_registry/certificate/routes.py:141]: Does a cancellation have to have a claimant? Not sure on the business logic here. How does this interface with the claim process?

File: gc_registry/certificate/services.py [hunk 2]
```diff
@@ -563,8 +562,12 @@ def query_certificates(
     if validate_query(certificate_query) is False:
         return None
 
-    # Query certificates based on the given filter parameters
-    stmt = select(GranularCertificateBundle)  # type: ignore
+    # Query certificates based on the given filter parameters, without returning deleted
+    # certificates
+    stmt = select(GranularCertificateBundle).where(
+        GranularCertificateBundle.account_id == certificate_query.source_id,
+        GranularCertificateBundle.is_deleted == False,  # noqa
```
Comment (ClemAtt) [inline gc_registry/certificate/services.py:569]: Should this be

`~GranularCertificateBundle.is_deleted`

File: gc_registry/certificate/routes.py (hunk 1 above)
Comment (CAGalbraith) [reply gc_registry/certificate/routes.py:141]: Claim definintely needs more fleshing out - I'm interpreting the standard as the gc bundle may be cancelled on behalf of the account or a third-party but it needs to be cancelled on behalf of someone

File: gc_registry/certificate/services.py (hunk 2 above)
Comment (CAGalbraith) [reply gc_registry/certificate/services.py:569]: Ruff seems to think so but in tariffscanner at least sqlmodel complains when passing boolean clauses through like that

**Pull request merged**